syntax = "proto3";

// The social network service definition.
service TSN {
	// Registers a new user
	rpc AddUser (UserRequest) returns (UserReply) {}
	rpc ListUsers (UserRequest) returns (ListUsersReply) {}
	rpc FollowUser (FollowUserRequest) returns (UserReply) {}
	rpc UnfollowUser (UnfollowUserRequest) returns (UserReply) {}
}

// The request message containing the user's name.
message UserRequest {
	string username = 1;
}

message FollowUserRequest {
	string username = 1;
	string user_to_follow = 2;
}

message UnfollowUserRequest {
	string username = 1;
	string user_to_unfollow = 2;
}

message ListUsersReply {
	int32 status = 1;
	string all_users = 2;
	string followed_users = 3;
}

// The response message containing the status (success or failure)
message UserReply {
	int32 status = 1;
}
